buildchain/buildchain/constants.py

```diff
@@ -63,7 +64,15 @@
 
 K8S_VERSION  : str = '1.11.7'
 SALT_VERSION : str = '2018.3.4'
+KEEPALIVED_VERSION : str = '1.3.5-8.el7_6'
```

Comment:
If those constants are only used in `image.py` then they should be defined there, closer to their uses.

Reply:
We're not using `SALT_VERSION` anywhere else either. It's a constant, over time this (and `SALT_VERSION` and whatever else) should go into some central location where Salt SLSs retrieve the values as well (I believe there's a ticket for that), so IMHO this makes sense?

Reply:
Ha yes, for the version indeed (so we can keep it here)

But the build ID and the SHA256 and the image name can be moved (like `SALT_MASTER_BUILD_ID`) or used inline in `image.py` (like the SHA256 and the image name) I think.

Reply:
Disagree.

`CENTOS_BASE_IMAGE` and `CENTOS_BASE_IMAGE_SHA256` are constants very much like `SALT_VERSION` which we may want to update from time to time, *without updating the implementation* of how things are built.

Moving `*_BUILD_ID` somewhere else would be counter-productive, because *build-id is to be reset to 1 when the version of something changes*.

Reply:
Then `SALT_MASTER_BUILD_ID` should be moved, to be consistent (and if we start having a lof of those version-related constant, we could consider having a dedicated file (`versions.py`) for them)

Reply:
No! The point is that `*_BUILD_ID` and `*_VERSION` *go hand in hand*, they're not two distinct unrelated entities. The *version* specifies the version of the software in the container. The *build* specifies the iteration of container building (basically, tracking changes to `Dockerfile`, whatever entrypoint script(s),...) which combined form the *tag* of a container image.

Furthermore, as I said before, *build* gets reset when *version* is bumped (RPM-style), so there again having them in separate places would be counter-productive/counter-intuitive/...

Yes, as I wrote elsewhere, version management must be more centralized (and cross the build-system / Salt boundary) to avoid duplication and divergence, but that's most definitely out of scope for this PR.

Reply:
> No! The point is that *_BUILD_ID and *_VERSION go hand in hand, they're not two distinct unrelated entities. 

Yeah, nothing in my message was against that (if it relates to my suggestion of moving version-related constants in their own file, this included the build ones, as they are tied to the version they are version-related).

But OK to do it in another PR, one day™